### PR TITLE
Changes for Nuage atomic ansible install

### PR DESCRIPTION
**Pull request by rparulek**
Changes for Nuage atomic ansible install

### CODE REVIEW

**Comment by rparulek** [comment]
Can any of the admins review and merge this into master?

**Comment by vareti** [comment]
aos-ci-test

**Comment by ingvagabund** [comment]
aos-ci-test

**Comment by rparulek** [comment]
@sdodson Can we merge this change set into master ?

**Comment by rparulek** [comment]
@michaelgugino This patch is trying to address a bunch of changes specific for Nuage
1) For atomic support with Nuage, adding a fix to do custom docker mounts of Nuage host dirs for atomic-openshift-node service to invoke Nuage CNI plugin to resolve pods on Atomic slave nodes if Nuage SDN is the selected vendor for ansible installation with Openshift.
2) Fixing a bug in Nuage nuage_master role code to assign proper value for the necessary variable
3) Fixing some redundant configuration in daemon sets for OSE master and slave nodes

**Comment by rparulek** [comment]
@michaelgugino Can we have this change set merged into master branch?

**Comment by abutcher** [comment]
aos-ci-test

**Comment by rparulek** [comment]
@abutcher Since I saw an error on the aos-ci run, I synced my branch to the latest master and rebased. This triggered a new run, my bad!

**Comment by rparulek** [comment]
@sdodson @abutcher Can we stage this changeset to be merged into master branch please?

**Comment by rparulek** [comment]
PR has been rebased and synced to the master. Can any of the admins approve and merge this change set please?

**Comment by rparulek** [comment]
@sdodson Can we get this change set merged into master branch?

File: roles/openshift_node/templates/openshift.docker.node.service [hunk 1]
```diff
@@ -34,6 +34,7 @@ ExecStart=/usr/bin/docker run --name {{ openshift.common.service_type }}-node \
   -v /lib/modules:/lib/modules -v /etc/origin/openvswitch:/etc/openvswitch \
   -v /etc/origin/sdn:/etc/openshift-sdn -v /var/lib/cni:/var/lib/cni \
   -v /etc/systemd/system:/host-etc/systemd/system -v /var/log:/var/log \
+  {% if openshift_use_nuage -%} $NUAGE_ADDTL_BIND_MOUNTS {% endif -%} \
```
Comment (sdodson) [inline roles/openshift_node/templates/openshift.docker.node.service]: Add a default here so if the variable is not defined then it defaults to false.
`+  {% if openshift_use_nuage | default(false,true) -%} $NUAGE_ADDTL_BIND_MOUNTS {% endif -%} \`

Comment (rparulek) [reply roles/openshift_node/templates/openshift.docker.node.service]: @sdodson Addressed the review comment. Thanks!

**Comment by rparulek** [comment]
@sdodson Does this change set look good to merge?

File: roles/openshift_node/templates/openshift.docker.node.service [hunk 2]
```diff
@@ -34,6 +34,7 @@ ExecStart=/usr/bin/docker run --name {{ openshift.common.service_type }}-node \
   -v /lib/modules:/lib/modules -v /etc/origin/openvswitch:/etc/openvswitch \
   -v /etc/origin/sdn:/etc/openshift-sdn -v /var/lib/cni:/var/lib/cni \
   -v /etc/systemd/system:/host-etc/systemd/system -v /var/log:/var/log \
+  {% if openshift_use_nuage | default(false) -%} $NUAGE_ADDTL_BIND_MOUNTS {% endif -%} \
```
Comment (michaelgugino) [inline roles/openshift_node/templates/openshift.docker.node.service]: I think '-%}' above may be in error.  No need for the '-' character in the if/endif blocks.

Comment (ingvagabund) [reply roles/openshift_node/templates/openshift.docker.node.service]: My 50c guess, the `-` will remove the extra NL which should have no effect on the one-line expression.

**Pull request merged**